app/src/main/java/com/github/pockethub/android/util/ImageBinPoster.kt

```diff
@@ -0,0 +1,73 @@
+package com.github.pockethub.android.util
+
+import android.content.Context
+import android.net.Uri
+import io.reactivex.Maybe
+import okhttp3.*
+import okio.Okio
+import java.io.IOException
+
+object ImageBinPoster {
+
+    /**
+     * Post the image to ImageBin
+     *
+     * @param context A context
+     * @param uri The content URI
+     * @param callback Request callback
+     * @return If the file was successfully retrieved
+     */
+    @JvmStatic
+    fun post(context: Context, uri: Uri): Maybe<Response> {
+        var bytes: ByteArray? = null
+
+        try {
+            val stream = context.contentResolver.openInputStream(uri)
+            if (stream != null) {
+                val source = Okio.source(stream)
+                bytes = Okio.buffer(source).readByteArray()
+            }
+        } catch (e: IOException) {
+            e.printStackTrace()
+            return Maybe.empty()
+        }
+
+        return post(bytes)
+    }
+
+    /**
+     * Post the image to ImageBin
+     *
+     * @param bytes Bytes of the image to post
+     * @param callback Request callback
+     */
+    @JvmStatic
+    fun post(bytes: ByteArray?): Maybe<Response> {
```

Comment:
Why are these `Maybe`?

Reply:
On failure I for some reason thought I should return `Maybe.empty()` instead of `Single.error()`. I'll amend this with a rebase.